### PR TITLE
test(combineAll, combineLatest operators): execute tests with TestScheduler#run

### PR DESCRIPTION
**Pull request by rall**
Use TestScheduler#run for the combineAll and combineLatest operator tests

### CODE REVIEW

File: spec/operators/combineAll-spec.ts [hunk 1]
```diff
@@ -1,453 +1,522 @@
...
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const e1 = cold('|');
+      const e1subs = ' (^!)';
+      const e2 = cold('|');
+      const e2subs = ' (^!)';
+      const expected = '|';
```
Comment (cartant) [inline spec/operators/combineAll-spec.ts]: I think `expected` was misaligned in the previous version. It should line up with the other bars.

File: spec/operators/combineAll-spec.ts [hunk 2]
```diff
@@ -1,453 +1,522 @@
...
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('-a-^-|');
+      const e1subs = '   ^-!';
+      const e2 = hot('------'); //never
+      const e2subs = '   ^--';
+      const expected = ' -'; //never
```
Comment (cartant) [inline spec/operators/combineAll-spec.ts]: If you want, you can extend the `-` in `expected` to line up with the end of `e2` - like in the test that follows this one. If the marble diagram has nothing but `-` in it, where is ends doesn't matter.

File: spec/operators/combineAll-spec.ts [hunk 3]
```diff
@@ -1,453 +1,522 @@
...
-    const rightSubs =      '^                   !';
-    const expected =       '-----------x--y--z--#';
-
-    const result = of(left, right).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected, { x: 'cd', y: 'ce', z: 'cf' }, 'dun dun dun');
-    expectSubscriptions(left.subscriptions).toBe(leftSubs);
-    expectSubscriptions(right.subscriptions).toBe(rightSubs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const left =  hot('--a--^--b--c--|');
```
Comment (cartant) [inline spec/operators/combineAll-spec.ts]: There are two spaces in `=  hot`, so Prettier will effect a misalignment here.

File: spec/operators/combineLatest-spec.ts [hunk 4]
```diff
@@ -1,480 +1,551 @@
...
+  });
+
   asDiagram('combineLatest')('should combine events from two cold observables', () => {
-    const e1 =   cold('-a--b-----c-d-e-|');
-    const e2 =   cold('--1--2-3-4---|   ');
-    const expected = '--A-BC-D-EF-G-H-|';
+    testScheduler.run(({ cold, expectObservable }) => {
+      const e1 = cold('  -a--b-----c-d-e-|');
+      const e2 = cold('  --1--2-3-4---|   ');
+      const expected = '--A-BC-D-EF-G-H-|';
```
Comment (cartant) [inline spec/operators/combineLatest-spec.ts]: I think the original version was misaligned (you will find a bit of that in the code base). All of the snippets should start at the same time.

File: spec/operators/combineLatest-spec.ts [hunk 5]
```diff
@@ -1,480 +1,551 @@
...
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const e1 = cold('|');
+      const e1subs = ' (^!)';
+      const e2 = cold('|');
+      const e2subs = ' (^!)';
+      const expected = '|';
```
Comment (cartant) [inline spec/operators/combineLatest-spec.ts]: `expected` in the original version was misaligned.

File: spec/operators/combineLatest-spec.ts [hunk 6]
```diff
@@ -1,480 +1,551 @@
...
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const values = {
+        a: 1
+      };
+      const e1 = hot('-a-^-|', values);
+      const e1subs = '   ^-!';
+      const e2 = hot('------', values); //never
+      const e2subs = '   ^--';
+      const expected = ' -'; //never
```
Comment (cartant) [inline spec/operators/combineLatest-spec.ts]: As in the next test, `expected` can end where `e2` ends, but it was no impact on the test.

**Comment by rall** [comment]
ah, sorry - I should have checked my work more carefully. I'll make these fixes now

File: spec/operators/combineAll-spec.ts (hunk 1 above)
Comment (rall) [reply spec/operators/combineAll-spec.ts]: I'd think we would want to align on the ^ in`(^!)`, but it looks like we align on the opening parenthesis in other examples so I'll do that here - is that right?

**Comment by rall** [comment]
All set I think. I'll look out for these issues in the other operator specs as I get them

File: spec/operators/combineAll-spec.ts (hunk 1 above)
Comment (cartant) [reply spec/operators/combineAll-spec.ts]: The `(^!)` is a subscription marble: `^` represents subscription and `!` represents unsubscription. The parentheses mean that the contents occur in the frame at which the `(` is placed, so, yeah, it needs to align with the `(`.

Note that `^` has a different meaning when it's used in an observable's marble diagram. There it represents the zero frame.

**Comment by benlesh** [comment]
YES! So good!

**Pull request merged**